lms/static/js/learner_analytics_dashboard/LearnerAnalyticsDashboard.jsx

```diff
@@ -0,0 +1,145 @@
+  const countsArray = types.map((type) => {
+    return {
+      [type]: countByType(type, assignments)
+    }
+  });
+
+  return arrayToObject(countsArray);
+}
+
+function getStreakIcons(count) {
+  return Array.apply(null, { length: count }).map((e, i) => (
+    <span className="fa fa-trophy" aria-hidden="true" key={i}></span>
+  ));
+}
+
+function getStreakString(count) {
+  const unit = (count ===1) ? 'week' : 'weeks';
+  return `Logged in ${count} ${unit} in a row`;
+}
+
+export function LearnerAnalyticsDashboard(props) {
+console.log('props: ', props);
+  const {grading_policy, grades, schedule, week_streak, weekly_active_users, discussion_info} = props;
+  // temp. for local dev
+  // const week_streak = 3;
+  // const weekly_active_users = 83400;
+  const gradeBreakdown = grading_policy.GRADER.map(({type, weight}, index) => {
+    return {
+      value: weight,
+      label: type,
+      sliceIndex: index + 1
+    }
+  }); //.sort((a, b) => a.value < b.value);
+
+  // Get a list of assignment types minus duplicates
+  const assignments = gradeBreakdown.map(value => value['label']);
+  const assignmentTypes = [...new Set(assignments)];
+  const assignmentCounts = getAssignmentCounts(assignmentTypes, schedule);
+  const tableData = [
+    {
+      label: 'Problem Set 1',
+      user: '12',
+      passing: '20',
+      total: '30'
+    },
+    {
+      label: 'Problem Set 2',
+      user: '20',
+      passing: '10',
+      total: '20'
+    },
+    {
+      label: 'Problem Set 3',
+      user: '0',
+      passing: '40',
+      total: '50'
+    }
+  ];
+
+  return (
+    <div className="learner-analytics-wrapper">
+      <div className="main-block">
+        <div className="analytics-group">
+          <h2 className="group-heading">Grading</h2>
+          <h3 className="section-heading">Weight</h3>
+          <div className="grading-weight-wrapper">
+            <div className="chart-wrapper">
+              <CircleChart
+                slices={gradeBreakdown}
+                centerHole={true}
+                sliceBorder={{
+                    strokeColor: '#f5f5f5',
+                    strokeWidth: 2
+                }}
+              />
+            </div>
+            <CircleChartLegend data={gradeBreakdown} />
+          </div>
+
+          <h3 className="section-heading">Graded Assessments</h3>
+          <div className="graded-assessments-wrapper">
+            <GradeTable assignmentTypes={assignmentTypes} grades={grades} />
+            <p className="footnote">*Calculated based on current average</p>
+          </div>
+        </div>
+        <div className="analytics-group">
+          <Discussions {...discussion_info} />
+        </div>
+      </div>
+      <div className="analytics-group sidebar">
+        <h2 className="group-heading">Timing</h2>
+        <h3 className="section-heading">Course due dates</h3>
+        <DueDates dates={schedule} assignmentCounts={assignmentCounts} />
+        {week_streak > 0 && 
```

Comment:
Do you think showing people that they've logged in 0 weeks in a row would encourage them to login with greater frequency?

Reply:
I will sync up with UX and see what they were thinking re. the behaviour here.

Reply:
I didn't look at the code, just this comment, but I would ensure that we display a minimum of 1 for week streak, because presumably if you are logged in and looking you have at least a streak of 1.